Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -253,57 +254,71 @@ class TimestampValue : public BaseValue {
   Timestamp value_;
 };
 
+/**
+ * Represents a locally-applied Server Timestamp.
+ *
+ * Notes:
+ *   - ServerTimestampValue instances are created as the result of applying an
+ *     FSTTransformMutation (see [FSTTransformMutation applyTo]). They can only
+ *     exist in the local view of a document. Therefore they do not need to be
+ *     parsed or serialized.
+ *   - When evaluated locally (e.g. via DocumentSnapshot data), they by default
+ *     evaluate to null.
+ *   - This behavior can be configured by passing custom FieldValueOptions to
+ *     `valueWithOptions:`.
```

Comment:
Nit: `valueWithOptions` looks like a remnant of the Objective-C version of this class.

Reply:
At this point in time this is correct, though it could be qualified with `FSTFieldValue`. Once #3178 lands this will need to refer to `-[FIRDocumentSnapshot convertedTimestamp:options:]`, if it refers to anything at all.